protostar/git/git_repository.py

```diff
@@ -120,122 +80,47 @@ def add_submodule(
+        self._git(["rm", "--force", str(submodule_path)])
 
-    @wrap_git_exception
-    def commit(self, msg: str):
-        credentials = [] if has_git_credentials() else DEFAULT_CREDENTIALS
-        subprocess.run(
-            ["git", *credentials, "commit", "-m", msg],
-            **SHARED_KWARGS,
-            cwd=self.repo_path,
-        )
+    def commit(self, msg: str) -> None:
+        self._git(["commit", "-m", msg])
 
-    @wrap_git_exception
-    def checkout(self, branch: str):
-        subprocess.run(["git", "checkout", branch], **SHARED_KWARGS, cwd=self.repo_path)
+    def checkout(self, branch: str) -> None:
+        self._git(["checkout", branch])
 
-    @wrap_git_exception
-    def create_tag(self, name: str):
-        credentials = [] if has_git_credentials() else DEFAULT_CREDENTIALS
-        subprocess.run(
-            ["git", *credentials, "tag", name], **SHARED_KWARGS, cwd=self.repo_path
-        )
+    def create_tag(self, name: str) -> None:
+        self._git(["tag", name])
 
-    @wrap_git_exception
     def get_tag_rev(self) -> str:
-        process = subprocess.run(
-            ["git", "rev-list", "--tags", "--max-count=1"],
-            stderr=subprocess.DEVNULL,
-            stdout=subprocess.PIPE,
-            check=True,
-            cwd=self.repo_path,
-        )
-        return process.stdout.decode("utf-8").strip()
+        return self._git(["rev-list", "--tags", "--max-count=1"])
 
-    @wrap_git_exception
     def get_tag(self, rev: Optional[str] = None) -> str:
-        process = subprocess.run(
-            ["git", "describe", "--tags"] + ([rev] if rev else []),
-            stderr=subprocess.DEVNULL,
-            stdout=subprocess.PIPE,
-            check=True,
-            cwd=self.repo_path,
-        )
-        return process.stdout.decode("utf-8").strip()
+        return self._git(["describe", "--tags"] + ([rev] if rev else []))
 
-    @wrap_git_exception
     def get_head(self) -> str:
-        process = subprocess.run(
-            ["git", "rev-parse", "HEAD"],
-            stderr=subprocess.DEVNULL,
-            stdout=subprocess.PIPE,
-            check=True,
-            cwd=self.repo_path,
-        )
-        return process.stdout.decode("utf-8").strip()
+        return self._git(["rev-parse", "HEAD"])
 
-    @wrap_git_exception
     def fetch_tags(self):
-        credentials = [] if has_git_credentials() else DEFAULT_CREDENTIALS
-        subprocess.run(
-            ["git", *credentials, "fetch", "--tags"],
-            **SHARED_KWARGS,
-            cwd=self.repo_path,
-        )
+        self._git(["fetch", "--tags"])
 
-    def get_submodules(self) -> Dict[str, Submodule]:
-        """
-        Returns a dictionary of form:
-        submodule_name: PackageInfo
-        """
+    def _git(self, args: list[str]) -> str:
+        return run_git(args=args, cwd=self.repo_path)
 
+    def get_submodule_name_to_submodule(self) -> dict[str, Submodule]:
```

Comment:
I actually liked the previous name better, or we could do `get_submodules_dict`? Submodule appears 2 time in the name and it bugs me 😬

Reply:
The previous name was better from the perspective of this file, but it was obscure for callers of that method because there was no info about what is the key without reading the comment.

`get_name_to_submodule`?

Reply:
`get_name_to_submodule` is fine with me, although usually i don't need that much verbosity in code, it actually obscures what is important in the function logic. this is my personal opinion though. Do as you please - just bringing this up for the sake of discussion and future consideration

Reply:
I agree that type provides enough verbosity

Reply:
1) The type doesn't say what is a string. You have to either guess it, have an explanatory comment, or read the code. 2) The name (`get_submodules`) suggests a list, which is misleading.
3) The bigger the scope, the more explicit the name should be.

Reply:
And this comment is an example why they should be avoided. The comment says the dictionary contains `PackageInfo` which is true.

Reply:
You can alias `SubmoduleName` as a `str` and the problem is solved.
When reading the code i did not understand what was happening in a function when given the new verbose name. Describing dictionaries with this verbosity in python is not a commonly seen pattern. 

Why have the type system if we don't use all of it, and keep relying on names to provide full information? 
Isn't the above information duplication as well? 

My mental process when wanting to use a method:
1. Check module/class name
2. Skim through available method names, to check how can i get what i want (submodules)
3. Check what it returns
4. Use it

Where do you usually go differently?

Reply:
[use implicit name](https://github.com/software-mansion/protostar/pull/1455/commits/5804133c45c09f8177acbb405b7e7db6bd950e64)